Frameworks/UIKit/UITextField.mm

```diff
@@ -322,43 +394,109 @@ - (NSDictionary*)typingAttributes {
 // Properties: Sizing the Text Field�s Text
 //
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)setAdjustsFontSizeToFitWidth:(BOOL)adjust {
-    UNIMPLEMENTED();
+    _adjustsFontSizeToFitWidth = adjust;
+    [self _adjustFontSizeToFitWidthOrApply];
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
-- (void)adjustsFontSizeToFitWidth:(BOOL)adjust {
-    UNIMPLEMENTED();
+- (BOOL)adjustsFontSizeToFitWidth {
+    return _adjustsFontSizeToFitWidth;
+}
+
+- (void)_adjustFontSizeToFitWidthOrApply {
+    _adjustedFont = self.font;
+
+    if (_adjustedFont == nil) {
+        _adjustedFont = [UIFont fontWithName:@"Segoe UI" size:[UIFont labelFontSize]];
+    }
+
+    // If the min size is greater than our font size, the font wins out.
+    // If the font's pointsize is smaller than the global min font size, the font wins out.
+    if (self.adjustsFontSizeToFitWidth && (self.minimumFontSize < _adjustedFont.pointSize) && (_adjustedFont.pointSize > g_minimumFontSize) && self.text && self.text.length) {
```

Comment:
Slight perf upgrade if you move self.text and self.text.length after self.adjustFontSizeToFitWidth. 

Could we find a way to simply the 5 conditions down or break it up?